plugins/ParaViewFidesEngine.cpp

```diff
@@ -0,0 +1,206 @@
+                                  : std::string("stub");
+        std::cout << "\tImplementation: " << implementation << "\n\n";
+    }
+
+    void CatalystInit()
+    {
+        conduit_cpp::Node node;
+        node["catalyst/scripts/script/filename"].set(this->ScriptFileName);
+
+        // options to set up the fides reader in paraview
+        std::ostringstream address;
+        address << &Io;
+
+        node["catalyst/fides/json_file"].set(this->JSONFileName);
+        node["catalyst/fides/data_source_io/source"].set(std::string("source"));
+        node["catalyst/fides/data_source_io/address"].set(address.str());
+        node["catalyst/fides/data_source_path/source"].set(
+            std::string("source"));
+        node["catalyst/fides/data_source_path/path"].set(
+            std::string("DataReader"));
+        catalyst_initialize(conduit_cpp::c_node(&node));
+
+        if (this->Rank == 0)
+        {
+            this->CatalystConfig();
+        }
+    }
+
+    void CatalystExecute()
+    {
+        auto timestep = this->Writer->CurrentStep();
+        conduit_cpp::Node node;
+        node["catalyst/state/timestep"].set(timestep);
+        // catalyst requires the next one, but when using Fides as the reader
+        // for Catalyst, it will grab the time from the correct adios variable
+        // if it is specified in the data model
+        node["catalyst/state/time"].set(timestep);
+        node["catalyst/channels/fides/type"].set(std::string("fides"));
+
+        // catalyst requires the data node on a channel, but we don't actually
+        // need it when using fides, so just create a dummy object to pass
+        // the validation in catalyst
+        conduit_cpp::Node dummy;
+        dummy["dummy"].set(0);
+        node["catalyst/channels/fides/data"].set(dummy);
+
+        catalyst_execute(conduit_cpp::c_node(&node));
+    }
+};
+
+ParaViewFidesEngine::ParaViewFidesEngine(adios2::core::IO &io,
+                                         const std::string &name,
+                                         adios2::helper::Comm comm)
+: adios2::plugin::PluginEngineInterface(io, name, adios2::Mode::Write,
+                                        comm.Duplicate()),
+  Impl(new EngineImpl(io.m_ADIOS))
+{
+    // Need to define the Variables in the IO object used for the inline engine
+    const auto &varMap = io.GetVariables();
+    for (const auto &it : varMap)
+    {
+        if (false)
+        {
+        }
+#define declare_type(T)                                                        \
+    else if (it.second->m_Type == adios2::helper::GetDataType<T>())            \
+    {                                                                          \
+        this->Impl->Io->DefineVariable<T>(                                     \
+            it.first, it.second->m_Shape, it.second->m_Start,                  \
+            it.second->m_Count, it.second->IsConstantDims());                  \
+    }
```

Comment:
So it's a little nitpicky, but the reason I have the if (false) is so that all the macro created sections are if else. That way when one is found, all the remaining if statements don't have to be evaluated. With your suggestion, every if statement would have to be evaluated every time this is called, even if the data type is the first one. This is similar to many other places in the code base. Though most other places have something like `if (type == adios2::DataType::Struct)` with an empty section.

Reply:
Ah completely missed the continue, and only looked at that snippet and forgot that it was even in a loop. :upside_down_face: